### PR TITLE
Zap logger

### PR DESCRIPTION
**Pull request by phudtran**
This PR adds the zap wrapper + units tests, and  the Merak Logger interface for the Merak Logger.
Capabilities include

- Different log levels
   - Info, Debug, Warn, Error, Panic, and Fatal
- Dynamic log level setting
- Log to syslog
- Log to file
- Log to stdout
- Log to both stdout and /var/log/merak/service_name

Users can create and use this logger as follows

```
// Logs will print to stdout and be stored under /var/log/merak/your_service_name_here
logger, err := NewLogger(INFO, "your_service_name_here")

logger.Info("The values are:", "a", "2")

// Once done sync the logs

logger.flush()
```

### CODE REVIEW

File: services/logger/zap.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+		core := zapcore.NewCore(encoder, zapcore.AddSync(f), opts.level)
+		return core, nil
+	case Stdout:
+		return zapcore.NewCore(encoder, os.Stdout, opts.level), nil
+	}
+	return nil, merakLogError{errors.New("invalid logger case"), strconv.Itoa(int(opts.location))}
+}
+
+// Creates a new logger that writes to stdout with the given log level
+func NewLogger(level Level) (*MerakLog, error) {
```
Comment (cj-chung) [inline services/logger/zap.go:102]: @phudtran So, we need to decide which logger (stdout, syslog, or file) we want to use in each service? Let's make our logger saving the log data into /var/log/merak for each service. Can you write an example to show how to use this logger to write the log data into a file and still can retrieve the log info from `kubectl logs` command.

File: services/logger/logger.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+	THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+	FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+	WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+package logger
+
+// Interface for the Merak Logger
+type Logger interface {
+	Flush() error
```
Comment (cj-chung) [inline services/logger/logger.go:18]: @phudtran Let's move the logger into /api as common lib.

File: services/logger/zap.go (hunk 1 above)
Comment (phudtran) [reply services/logger/zap.go]: Sure, I've updated the PR and changed the default logger to log both to /var/log/merak/service_name and stdout/console which will be visible from kubectl logs

File: services/logger/logger.go (hunk 2 above)
Comment (phudtran) [reply services/logger/logger.go]: Ok I've moved this to the common folder

File: services/logger/zap.go (hunk 1 above)
Comment (phudtran) [reply services/logger/zap.go]: Also updated the PR with an example.

**Review by cj-chung — APPROVED** [review]
LGTM

**Pull request merged**